### PR TITLE
make github CLI work in Git Bash

### PR DESCRIPTION
**Pull request by shiftkey**
With the current latest release (but probably earlier builds), `github .` won't work in Git Bash:

This is because the bash environment won't resolve a `.cmd` or `.bat` file matching the name like `cmd.exe` and Powershell do.

The fix for this is to bundle a second trampoline and script pair that light up in Git Bash. Given how similar this is to the current batch trampoline, I've borrowed a bunch of that code here.

 - [x] review changes and ensure it's good to ship
 - [x] reduce duplicate code for generating trampoline
 - [x] test `cmd` and `powershell` are unaffected

### CODE REVIEW

**Comment by joshaber** [comment]
>The fix for this is to bundle a second trampoline

giphy-2

**Comment by shiftkey** [comment]
This is ready to be reviewed and merged as part of the post 1.0 tidy up 🎉

File: app/src/main-process/squirrel-updater.ts [hunk 1]
```diff
@@ -72,33 +79,65 @@ function getBinPath(): string {
...
+  const versionedPath = resolveVersionedPath(
+    binPath,
+    'resources/app/static/github.sh'
+  )
+  const trampoline = `#!/usr/bin/env bash
+  DIR="$( cd "$( dirname "\$\{BASH_SOURCE[0]\}" )" && pwd )"
+  sh "$DIR/${versionedPath}" "$@"`
+  const trampolinePath = Path.join(binPath, 'github')
+  return new Promise<void>((resolve, reject) => {
+    Fs.ensureDir(binPath, err => {
```
Comment (joshaber) [inline app/src/main-process/squirrel-updater.ts]: Should we use `mkdirIfNeeded` from `file-system` instead here?

Comment (shiftkey) [reply app/src/main-process/squirrel-updater.ts]: Heck yeah we should! :metal:

**Comment by shiftkey** [comment]
🚚

File: app/src/main-process/squirrel-updater.ts [hunk 2]
```diff
@@ -72,30 +80,52 @@ function getBinPath(): string {
...
   )
-  const trampline = `@echo off\n"%~dp0\\${versionedPath}" %*`
+
+  const trampoline = `@echo off\n"%~dp0\\${versionedPath}" %*`
   const trampolinePath = Path.join(binPath, 'github.bat')
-  return new Promise<string>((resolve, reject) => {
-    Fs.ensureDir(binPath, err => {
+
+  return new Promise<void>((resolve, reject) => {
+    Fs.writeFile(trampolinePath, trampoline, err => {
```
Comment (joshaber) [inline app/src/main-process/squirrel-updater.ts]: (I should have raised this previously, sorry :hurtrealbad:)

I think we should make an async `writeFile` (or whatever) in `file-system`. I bet we're using this promise wrapping in a number of places at this point.

Comment (shiftkey) [reply app/src/main-process/squirrel-updater.ts]: All good. Yeah, it'll definitely get some reuse elsewhere.

**Comment by shiftkey** [comment]
:moon:

**Pull request merged**